crates/common_utils/src/types.rs

```diff
@@ -255,6 +325,64 @@ impl MinorUnit {
     pub fn new(value: i64) -> Self {
         Self(value)
     }
+
+    /// Convert the amount to its major denomination based on Currency and return String
+    pub fn to_major_unit_as_string(
+        &self,
+        currency: enums::Currency,
+    ) -> Result<StringMajorUnit, TryFromIntError> {
+        let amount_f64 = self.to_major_unit_asf64(currency)?;
+        let amount_string = format!("{:.2}", amount_f64.0);
+        Ok(StringMajorUnit::new(amount_string))
+    }
+
+    /// Convert the amount to its major denomination based on Currency and return f64
+    pub fn to_major_unit_asf64(
+        &self,
+        currency: enums::Currency,
+    ) -> Result<FloatMajorUnit, TryFromIntError> {
+        let amount_f64: f64 = u32::try_from(self.0)?.into();
+        let amount = if currency.is_zero_decimal_currency() {
+            amount_f64
+        } else if currency.is_three_decimal_currency() {
+            amount_f64 / 1000.00
+        } else {
+            amount_f64 / 100.00
+        };
+        Ok(FloatMajorUnit::new(amount))
+    }
+
+    ///Convert the higher decimal amount to its major absolute units
+    pub fn to_minor_unit_as_string(
+        &self,
+        currency: enums::Currency,
+    ) -> Result<String, ParseFloatError> {
+        let amount_f64 = self.0.to_string().parse::<f64>()?;
+        let amount_string = if currency.is_zero_decimal_currency() {
+            amount_f64
+        } else if currency.is_three_decimal_currency() {
+            amount_f64 * 1000.00
+        } else {
+            amount_f64 * 100.00
+        };
+        Ok(amount_string.to_string())
+    }
```

Comment:
You should define the input type, so that it can be only called when input is f64